homeassistant/components/unifiprotect/media_player.py

```diff
@@ -131,6 +131,28 @@ async def async_media_stop(self) -> None:
             await self.device.stop_audio()
             self._async_updated_event(self.device)
 
+    @callback
+    def _make_local_url(self, media_url: str) -> str:
+        # ffmpeg streams audio from current machine
+        # so it should not route outside of localhost
+        host = "localhost"
+        scheme = "https" if self.hass.http.ssl_certificate else "http"
```

Comment:
Maybe borrow something like this https://github.com/home-assistant/core/blob/117ab4a0e5b319fee7133ad8b8278e1355f3938a/homeassistant/components/onvif/event.py#L569

And than use YARL to construct a new url